File: Programmers/pro_67257_수식 최대화.py
def calPriority(split, priority):
	stack=split
	for i in priority:
		split = stack
		stack=[]
		j=0
		while(j<len(split)):
			if(i==split[j]):
				if(i=='*'):
					stack[-1]*=split[j+1]
				elif(i=='-'):
					stack[-1]-=split[j+1]
				elif(i=='+'):  
					stack[-1]+=split[j+1]
				j+=1
			else:
				stack.append(split[j])
			j+=1
	return abs(stack[0])
    
def solution(expression):
    priority=['-+*','-*+','+-*','+*-','*+-','*-+']
    split = []
    cur=''
    for i in expression:
        if(i in '-+*'):
            split+=[int(cur),i]
            cur=''
        else:
        	cur+=i
    split.append(int(cur))
        
    answer = 0
    for i in range(6):
    	answer=max(calPriority(split, priority[i]),answer)
    return answer

File: Programmers/pro_67257_수식 최대화.js
function solution(expression) {
    expression = expression.split(/([\*\+-])/g)
    const prio = ["*+-", "*-+", "+*-", "+-*", "-+*", "-*+"]
    let answer = 0;
    for (let i = 0; i < prio.length; i++) {
        let ret = Array.from(expression)
        for (let j = 0; j < 3; j++) {
            let operator = prio[i][j]
            let idx = ret.indexOf(operator)
            while (idx != -1) {
                let elem = 0
                if (operator == "*")
                    elem = ret[idx - 1] * ret[idx + 1]
                else if (operator == "+")
                    elem = ret[idx - 1] * 1 + ret[idx + 1] * 1
                else
                    elem = ret[idx - 1] * 1 - ret[idx + 1] * 1
                ret.splice(idx - 1, 3, elem)
                idx = ret.indexOf(operator)
            }
        }
        answer = Math.max(answer, Math.abs(ret))
    }
    return answer;
}